pages/workoutlist.js

```diff
@@ -0,0 +1,176 @@
+import { useState } from "react";
+import styled from "styled-components";
+import { exercises } from "@/lib/exercises";
+import { workouts } from "@/lib/workouts";
+
+function findExerciseById(exerciseId) {
+  return exercises.find((exercise) => exercise.id === exerciseId);
+}
+
+export default function WorkoutsList() {
+  const [showDetails, setShowDetails] = useState({});
+
+  const toggleDetails = (workoutId) => {
+    setShowDetails((prev) => ({
+      ...prev,
+      [workoutId]: !prev[workoutId],
+    }));
+  };
```

Comment:
Please maintain consistency by writing functions using the `function` keyword. Additionally, I recommend storing the IDs of the visible workouts in an array instead of keeping a boolean for each workout. This approach simplifies the state management and makes the code more efficient and easier to understand.

```suggestion
    const [visibleWorkouts, setVisibleWorkouts] = useState([]);

function toggleDetails(workoutId) {
    setVisibleWorkouts((prevVisibleWorkouts) => {
      if (prevVisibleWorkouts.includes(workoutId)) {
        return prevVisibleWorkouts.filter((id) => id !== workoutId);
      } else {
        return [...prevVisibleWorkouts, workoutId];
      }
    });
  }
```

The `toggleDetails` function manages the visibility state of workout details. It checks if a specific workout ID (`workoutId`) is already present in the `visibleWorkouts` array, which stores the IDs of workouts currently shown on the screen.

- If the workout ID is already in the array, it removes the ID, effectively hiding the workout details.
- If the workout ID is not in the array, it adds the ID, making the workout details visible.